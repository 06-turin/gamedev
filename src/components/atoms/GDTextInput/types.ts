import { InputHTMLAttributes } from 'react';

export type GDTextInputProps = InputHTMLAttributes<HTMLInputElement>
