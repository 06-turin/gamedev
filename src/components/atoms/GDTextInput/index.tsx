export { GDTextInputComponent as GDTextInput } from './GDTextInput';
