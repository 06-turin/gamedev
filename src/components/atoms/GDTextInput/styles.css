.input-label {
  color: var(--mainTextColor);
  margin-bottom: 18px;
}

.input-label_invalid {
  color: var(--dangerColor);
}

.input-field_standard {
  background-color: transparent;
  margin-left: 12px;
  box-shadow: none;
  border: none;
  border-bottom: 2px solid;
  color: var(--mainTextColor);
  border-color: var(--mainTextColor);
  text-transform: uppercase;

  &:focus-visible {
    outline: none;
  }
}

.input-field_invalid {
  color: var(--dangerColor);
  border-color: var(--dangerColor);
}
