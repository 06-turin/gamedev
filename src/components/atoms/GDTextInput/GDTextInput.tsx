import './styles.css';
import React, { ChangeEventHandler, FC, FocusEventHandler } from 'react';
import classnames from 'classnames';

export type GDTextInputProps = {
  id: string
  title: string
  className?: string
  placeholder?: string
  value?: string
  onChange?: ChangeEventHandler<HTMLInputElement>
  onBlur?: FocusEventHandler<HTMLInputElement>
}

export const GDTextInput: FC<GDTextInputProps> = ({
  className, placeholder, title, id, value, onChange, onBlur,
}) => {
  const titleString = `${title}:`;
  return (
    <label htmlFor={id} className={classnames('input-label')}>
      {titleString}
      <input
        placeholder={placeholder}
        className={classnames(['input-field_standard', className])}
        type="text"
        id={id}
        name={title}
        value={value}
        onChange={onChange}
        onBlur={onBlur}
      />
    </label>
  );
};
