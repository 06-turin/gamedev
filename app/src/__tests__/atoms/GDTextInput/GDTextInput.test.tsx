import React from 'react';
import classNames from 'classnames';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { GDTextInput } from 'components/atoms/GDTextInput/GDTextInput';
import '@testing-library/jest-dom/extend-expect';

const mockFunction = () => 'returns stuff';

describe('GDTextInput', () => {
  it('renders with default props', () => {
    const tree = renderer
      .create(
        <GDTextInput id="name" title="name" />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with all props passed', () => {
    const tree = renderer
      .create(
        <GDTextInput
          id="name"
          title="name"
          name="name"
          type="text"
          className={classNames('class-name')}
          placeholder="placeholder value"
          value="value"
          onChange={mockFunction}
          onBlur={mockFunction}
          isInvalid={false}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with the right title', () => {
    const { getByText } = render(<GDTextInput id="id" title="title" />);
    expect(getByText('title:')).toBeInTheDocument();
  });
});
