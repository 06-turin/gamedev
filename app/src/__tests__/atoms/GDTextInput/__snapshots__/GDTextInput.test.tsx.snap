// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GDTextInput renders with all props passed 1`] = `
<label
  className="input-label"
  htmlFor="name"
>
  name:
  <input
    className="input-field_standard class-name"
    id="name"
    name="name"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="placeholder value"
    type="text"
    value="value"
  />
</label>
`;

exports[`GDTextInput renders with default props 1`] = `
<label
  className="input-label"
  htmlFor="name"
>
  name:
  <input
    className="input-field_standard"
    id="name"
    name="name"
    type="text"
  />
</label>
`;
